### PR TITLE
Account module

### PR DESCRIPTION
**Pull request by delfipolito**
screen

### CODE REVIEW

File: src/components/Account/AccountModule.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      // reject closing the popover
+      return false
+    }
+    setOpened(false)
+  }, [screenId])
+
+  return (
+    <div
+      ref={buttonRef}
+      tabIndex="0"
```
Comment (andy-hook) [inline src/components/Account/AccountModule.js:83]: Do we need this? it seems to interfere with keyboard tabbing, but I'm curious if it exists to fix some other issue 🤔

File: package.json [hunk 2]
```diff
@@ -49,7 +50,8 @@
     "react-router-dom": "^5.2.0",
     "react-spring": "^8.0.1",
     "styled-components": "^5.1.1",
-    "use-viewport": "^0.2.0"
+    "use-viewport": "^0.2.0",
+    "use-wallet": "^0.8.0"
```
Comment (andy-hook) [inline package.json:54]: Let's make sure to commit an updated lockfile for these new dependencies :)

**Review by andy-hook — COMMENTED** [review]
Looking super solid! 🚀 🚀 my main comments are about the network configuration which we already have available :)

File: src/components/Account/AccountButton.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  EthIdenticon,
+  IconDown,
+  GU,
+  RADIUS,
+  textStyle,
+  useTheme,
+  useLayout,
+} from '@aragon/ui'
+import { useWallet } from 'use-wallet'
+import { shortenAddress } from '../../lib/web3-utils'
```
[the remark this answers is not in the document]
Comment (delfipolito) [reply src/components/Account/AccountButton.js:14]: Thats supper weird! I swear it was there haha. I added it now :) thanks for the catch

File: src/components/Account/ScreenConnected.js [hunk 4]
```diff
@@ -80,13 +80,12 @@ function AccountScreenConnected({ wallet }) {
...
             <IdentityBadge
               entity={wallet.account}
-              label={shortenAddress(wallet.account)}
               compact
               badgeOnly
               css={`
                 cursor: pointer;
               `}
-              networkType={getNetworkType()}
+              networkType={getNetworkName()}
```
Comment (andy-hook) [inline src/components/Account/ScreenConnected.js]: We should pass the `chainId` in here to get the name

File: src/lib/web3-utils.js [hunk 5]
```diff
@@ -62,3 +62,17 @@ export function addressesEqual(first, second) {
   return first === second
 }
 
+export function shortenAddress(address, charsLength = 4) {
```
Comment (andy-hook) [inline src/lib/web3-utils.js:53]: Let's add the docs annotation for this util as we already have one available

File: yarn.lock [hunk 6]
```diff
@@ -64,6 +64,15 @@
     "@aragon/connect-ethereum" "^0.7.0-beta.3"
     "@aragon/connect-thegraph" "^0.7.0-beta.3"
 
+"@aragon/provided-connector@^6.0.7":
```
Comment (andy-hook) [inline yarn.lock:67]: Just checking – Are we definitely updating the existing lock file? the diff just looks a little large so want to make sure!

**Review by andy-hook — COMMENTED** [review]
A couple more little things ✨  I'm also going to check what's happening with the bundle size, seems a bit large.

File: yarn.lock (hunk 6 above)
Comment (andy-hook) [reply yarn.lock:67]: Never mind, use-wallet seems to be bring plenty of friends 😅

Comment (delfipolito) [reply yarn.lock:67]: yes :)

File: src/components/Account/ScreenConnected.js (hunk 4 above)
Comment (delfipolito) [reply src/components/Account/ScreenConnected.js]: This is outdated, now the code looks like this: `networkType={networkEnvironment.legacyNetworkType}` :)

**Review by andy-hook — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**